services/elasticache/pkg/resource/snapshot/custom_update_api.go

```diff
@@ -0,0 +1,16 @@
+package snapshot
+
+import (
+	"context"
+	ackcompare "github.com/aws/aws-controllers-k8s/pkg/compare"
+)
+
+// Snapshot API has no update
+func (rm *resourceManager) customUpdateSnapshot(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+	diffReporter *ackcompare.Reporter,
+) (*resource, error) {
+	return desired, nil
+}
```

Comment:
Is the above necessary? Is it not possible to stick with the default implementation fallthrough that returns `nil, ackerr.NotImplemented` when there is no Update operation for a resource?

Reply:
After creating snapshot from source snapshot we would populate `ReplicationGroupId` thus ACK would keep on invoking `sdkUpdate` and this keeps throwing `ackerr.NotImplemented`. Since there is no update snapshot API I think it should not be `ackerr.NotImplemented`?

Reply:
I don't understand why ACK would keep on invoking `sdkUpdate`. If `ReplicationGroupId` is returned from the call to `CreateSnapshot`, then the `Status.ReplicationGroupID` field of the CR should be set in the `GoCodeSetOutput` returned Go code, right? And that CR is then saved to etcd and there would not be a new reconcile loop for the snapshot CR.

Reply:
`Spec` is not patched so `desired` will not have `ReplicationGroupId` updated thus we see `sdkUpdate` invoked continuously.